### PR TITLE
Feature/ethereum like

### PR DESCRIPTION
**Pull request by wuminzhe**

### CODE REVIEW

File: ethereum/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+[package]
+name = "ethereum"
```
Comment (HackFisher) [inline ethereum/Cargo.toml:2]: Better to use another name for this crate

**Review by HackFisher — CHANGES_REQUESTED** [review]

File: ethereum/Cargo.toml (hunk 1 above)
Comment (wuminzhe) [reply ethereum/Cargo.toml:2]: Yes

Comment (HackFisher) [reply ethereum/Cargo.toml:2]: How about ethereum-chains for the crate name?

**Review by HackFisher — COMMENTED** [review]
There are Heco chain types and services added, are they ready for review and merge?

@songtianyi @xiaoch05 Please take look at this PR.

**Comment by wuminzhe** [comment]
This PR was paused for we start to use parity common bridges.

File: ethereum/src/ethereum_tracker.rs [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+use std::marker::PhantomData;
+use web3::Web3;
+use web3::transports::Http;
+
+pub struct EthereumLikeChainTracker<C: Chain, H: LogsHandler> {
+	client: EthereumLikeChainClient,
+	topics_list: Vec<(H160, Vec<H256>)>,
+	logs_handler: H,
+	from: u64,
+	stop: bool,
```
Comment (songtianyi) [inline ethereum/src/ethereum_tracker.rs:17]: rename `stop` to `run_once`

File: ethereum/src/logs_handler.rs [hunk 3]
```diff
@@ -0,0 +1,21 @@
+use crate::{
+	Result,
+	EthereumLikeChainClient
+};
+use async_trait::async_trait;
+use web3::types::{Log, H160, H256};
+
+#[async_trait]
+pub trait LogsHandler {
```
Comment (songtianyi) [inline ethereum/src/logs_handler.rs:9]: Rename `LogsHandler` to `ChainLogHandler`. Cause `LogsHandler`  not telling what kind of log we are dealing with.

File: ethereum/src/logs_handler.rs [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+	async fn handle(&self, client: &EthereumLikeChainClient, topics_list: &Vec<(H160, Vec<H256>)>, logs: Vec<Log>) -> Result<()>;
```
Comment (songtianyi) [inline ethereum/src/logs_handler.rs:10]: Rename `topics_list` to `topics`, just like `logs` did. We haven't use logs_list right?

File: src/cmd/heco.rs [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+		Http::new(&config.ethereum.rpc)
+			.map_err(|e| Error::NewHttpError(config.ethereum.rpc.clone(), e.to_string()))?,
+	);
+	let darwinia = darwinia_api::get_darwinia_instance(&config).await?;
+
+	let mut tracker = HecoLogsHandler::new(
+		config.clone(),
+		web3.clone(),
+		data_dir.clone(),
+		4006177,
```
Comment (songtianyi) [inline src/cmd/heco.rs:38]: Should we add a const for this magic number?

**Comment by HackFisher** [comment]
@fewensa Please also take a look at this pull request, are we good to have this in our next refactor, or put off/close it.

**Pull request closed without merging**